demos/random-values-stream/index.js

```diff
@@ -0,0 +1,217 @@
+                this.startValueInterval(controller);
+            }
+        },
+
+        async close(controller) {
+            logStatusText('`close` method of the readable stream called');
+            this.clearValueInterval();
+            controller.close();
+            return;
+        },
+
+        async cancel() {
+            logStatusText('`cancel` method of the readable stream called')
+            this.clearValueInterval();
+        },
+
+        throwFinalError(error) {
+            console.log('Errored out');
+            console.error(error);
+
+            this.clearValueInterval();
+        },
+
+        startValueInterval(controller) {
+            if (this.interval) {
+                return;
+            }
+
+            this.interval = setInterval(() => {
+                try {
+                    controller.enqueue(randomValuesUint8Array(20));
+                    this.totalEnqueuedItemsCount++;
+                    this.checkBackpressureSignal(controller);
+                    
+                    // Close the stream and clear the interval
+                    if (maxValues && this.totalEnqueuedItemsCount >= maxValues) {
+                        return this.close(controller);  
+                    }
+                } catch (error) {
+                    this.throwFinalError(error);
+                }
+            }, valueInterval);
+        },
+        
+        /**
+         * Clears the value interval stored in this.interval reference
+         */
+        clearValueInterval() {
+            if (this.interval) {
+                clearInterval(this.interval);
+                this.interval = null;
+            }
+        },
+
+        /**
+         * Checks a backpressure signal  and clears the interval
+         * not enqueuing any more values
+         * 
+         * @param {*} controller 
+         * @param {*} interval
+         */
+        checkBackpressureSignal(controller, interval) {
+            if (controller.desiredSize <= 0) {
+                this.clearValueInterval();
+            }
+        }
+    }, cqs);
+
+    return readableStream;
+}
+
+/**
+ * Creates a random values Uint8Array
+ * @param {*} numberOfBytes 
+ */
+function randomValuesUint8Array(numberOfBytes) {
+    const uint8Array = new Uint8Array(numberOfBytes);
+    return window.crypto.getRandomValues(uint8Array);
+}
+
+/**
+ * Create a writable stream to display the output with a builtin backpressure
+ */
+function createOutputWritableStream(parentElement) {
+    /**
+     * Equivalent to
+     * 
+     * const cqs = new CountQueuingStrategy({
+     *  highWaterMark: 3,
+     * });
+     */
+    const queuingStrategy = {
+        highWaterMark: 3,
+        size() { return 1; }
+    }
+
+    const writable = new WritableStream({
+        async write(chunk, controller) {
+            try {
+                await writeChunk(chunk);
+                return;
+            } catch (error) {
```

Comment:
Not necessary to catch the exception. If write() throws, the stream will errored automatically.